salt/states/libcloud_dns.py

```diff
@@ -71,7 +72,7 @@ def state_result(result, message, name, changes=None):
             'changes': changes}
 
 
-def zone_present(domain, type, profile):
+def zone_present(domain, profile, type=None, ttl=None, extra=None):
```

Comment:
name is a required argument for all state functions, i would just have name as the first argument, and then add domain on at the end to keep people be able to use domain: in their states, and if domain is set, use that instead of name.